### PR TITLE
Refactor for lesions analysis - Connectomics

### PR DESCRIPTION
**Pull request by frheault**
Similarly to tractometry for lesiosn, this give access to 3 news matrices for lesions. If the users provided a lesion mask, access to
- lesion_mask_mni_vol (total volume)
- lesion_mask_mni_count (lesion_count)
- lesion_mask_mni_sc (nbr of streamlines going throught the lesions)

To keep it simple, they are computed in scil_compute_connectivity.py and considered like all the other matrices.
At the moment, the lesions must be provided already in diffusion space. This is due to the fact that lesions can come from Flair, T1, T2, etc. and it is much more simple to request to move it to Diff space. This is compatible with the tractometry-flow, which also required lesions in the same space as metrics/bundles which is Diff space too.

PS: lesion_mask_mni_sc is like a disconnectome, the sc.npy is the full connectivity matrix and this one is the results after removing lesions. The --lesion_load make the compute_connectivity even slower, this is due to simple 0.1sec operation being done Nx(N-1)/2 time, which can add up to 20 minutes even for a simple Freesurfer.
If you activate EVERYTHING (commit, similarity, AFD, a bunch of metrics, lesions, with Brainnetome (246 regions) this single step for a single subject is like 2h+

This is an example of lesions count matrix
image

### CODE REVIEW

**Review by mdesco — COMMENTED** [review]
I now have tested and commented all the new scripts and I like a lot! After the minor comestic edits and answers to my questions, I can approve.

File: USAGE [hunk 1]
```diff
@@ -54,6 +54,7 @@ DESCRIPTION
                                                 │   │── *dwi.bvec (optional)
                                                 │   │── *peaks.nii.gz (optional)
                                                 │   │── *fodf.nii.gz (optional)
+                                                │   │── *lesion_mask.nii.gz (optional, diff space)
```
[the remark this answers is not in the document]
Comment (frheault) [reply USAGE:61]: @mdesco It is now a parameter

File: USAGE [hunk 2]
```diff
@@ -35,6 +35,10 @@ Which is (NBR_PARCEL) * (NBR_PARCEL-1) / 2. Running connectoflow with a freesurf
 parcellation is A LOT faster than with Brainnetome, which is A LOT faster than
 with Glasser, make sure to test & optimize calls before launching on clusters.
 
+If providing a lesion mask, the nifti should be a uint8 binary image representing
+the voxel that are considered lesions. These lesions should have been taken into account
+when generating the tractogram (to allow tractography within the lesions).
+
```
Comment (mdesco) [inline USAGE:41]: While we are on this, can you add a word on the labels from freesurfer or any parcellation in lines 5-8. To make sure the user realizes that the atlas must contain only non-zero labels of the connectome, i.e. strip out the wm labels, csf labels or any labels that are not cortical/subcortical and desired in the connectome.

File: USAGE [hunk 3]
```diff
@@ -35,6 +35,10 @@ Which is (NBR_PARCEL) * (NBR_PARCEL-1) / 2. Running connectoflow with a freesurf
 parcellation is A LOT faster than with Brainnetome, which is A LOT faster than
 with Glasser, make sure to test & optimize calls before launching on clusters.
 
+If providing a lesion mask, the nifti should be a uint8 binary image representing
+the voxel that are considered lesions. These lesions should have been taken into account
+when generating the tractogram (to allow tractography within the lesions).
```
Comment (mdesco) [inline USAGE:40]: Please explicitly mention that lesion_mask must be in the same space as the diffusion data (dwi, fodf, tractogram, ...).

**Review by mdesco — COMMENTED** [review]
If we test the pipeline with 2-3 subjects without turning off the similarity option, the Compute_Connectivity_with_similiarity fails (of course) but the pipeline terminates without an error and without the connectivity matrices... Can we robustify this?

**Review by mdesco — APPROVED** [review]
@frheault working great! Tested on 3 MS subjects and everything seems to do what it should.

Last request before merging: I find it very hard to extract the edge_keys from the visualize connectivity histogram. Any chance we can have:
i) a txt file output of edge pairs and/or nodes touched by lesions sorted by sc or vol?

ii) Also, talking to Sami, I think it would be cool to have a txt with percent coverage of each label of the atlas.

This could come in this PR or a small followup PR. I let you and @arnaudbore decide.

GTG!

**Comment by frheault** [comment]
@mdesco For both of these thing it would be a separate PR.
For the i) that would be only a few line of code to read a label list and an numpy array and for ii) it would have to be a split script scil_analyse_bundle_lesion_load.py (already merged) and scil_analyse_atlas_lesion_load.py

**Pull request merged**